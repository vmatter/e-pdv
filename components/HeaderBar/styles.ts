import styled from '@emotion/styled';
import { alpha } from '@material-ui/core/styles';
import InputBase from '@material-ui/core/InputBase';
import IconButton from '@material-ui/core/IconButton';

export const FlexGrow = styled.div`
  flex-grow: 1;
`;

export const MenuButton = styled(IconButton)`
  margin-right: ${({ theme }) => theme.spacing(2)};
`;

export const LogoWrapper = styled.a`
  height: 100%;
`;

export const LogoImg = styled.img`
  height: 45px;
  cursor:pointer;
`;

export const Search = styled.div`
  position: relative;
  width: 100%;
  border-radius: ${({ theme }) => theme.shape.borderRadius}px;
  background-color: ${({ theme }) => alpha(theme.palette.common.white, 0.15)};
  &:hover {
    background-color: ${({ theme }) => alpha(theme.palette.common.white, 0.25)};
  }
  margin-right: ${({ theme }) => theme.spacing(2)};
  margin-left: 0;

  ${({ theme }) => `${theme.breakpoints.up('sm')} {
    margin-left: ${theme.spacing(2)};
    width: auto;
  }`}
`;

export const SearchIconWrapper = styled.div`
  padding: ${({ theme }) => theme.spacing(0, 2)};
  height: 100%;
  position: absolute;
  pointer-events: none;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const CustomInput = styled(InputBase)`
  color: inherit;
  padding: ${({ theme }) => theme.spacing(1, 1, 1, 0)};
  padding-left: ${({ theme }) => `calc(1em + ${theme.spacing(4)})`};
  transition: ${({ theme }) => theme.transitions.create('width')};
  width: 100%;

  ${({ theme }) => `${theme.breakpoints.up('md')} {
  width: 20ch
  }`}
`;
