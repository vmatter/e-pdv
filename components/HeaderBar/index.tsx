import { useState, MouseEvent } from 'react';
import Link from 'next/link';
import { useRouter } from 'next/router';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import IconButton from '@material-ui/core/IconButton';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import MenuIcon from '@material-ui/icons/Menu';
// import SearchIcon from '@material-ui/icons/Search';
import AccountCircle from '@material-ui/icons/AccountCircle';
import MoreIcon from '@material-ui/icons/MoreVert';
import {
  FlexGrow,
  LogoImg,
  LogoWrapper,
  // Search,
  // SearchIconWrapper,
  // CustomInput,
  MenuButton,
} from './styles';
import { DesktopSection, MobileSection } from '../Layout/styles';
import { setAccessToken } from '../../utils/token';

const HeaderBar = () => {
  const router = useRouter();
  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
  const [
    mobileMoreAnchorEl,
    setMobileMoreAnchorEl,
  ] = useState<null | HTMLElement>(null);

  const isMenuOpen = Boolean(anchorEl);
  const isMobileMenuOpen = Boolean(mobileMoreAnchorEl);

  const handleProfileMenuOpen = (event: MouseEvent<HTMLElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleMobileMenuClose = () => {
    setMobileMoreAnchorEl(null);
  };

  const handleMenuClose = () => {
    setAnchorEl(null);
    handleMobileMenuClose();
  };

  const handleLogout = () => {
    handleMenuClose();
    setAccessToken('');
    router.push('/login');
  };

  const hendleProducts = () => {
    handleMenuClose();
    router.push('/products');
  };

  const handleMobileMenuOpen = (event: MouseEvent<HTMLElement>) => {
    setMobileMoreAnchorEl(event.currentTarget);
  };

  const menuId = 'primary-search-account-menu';
  const renderMenu = (
    <Menu
      anchorEl={anchorEl}
      anchorOrigin={{
        vertical: 'top',
        horizontal: 'right',
      }}
      id={menuId}
      keepMounted
      transformOrigin={{
        vertical: 'top',
        horizontal: 'right',
      }}
      open={isMenuOpen}
      onClose={handleMenuClose}
    >
      <MenuItem onClick={handleMenuClose}>Usuários</MenuItem>
      <MenuItem onClick={hendleProducts}>Produtos</MenuItem>
      <MenuItem onClick={handleLogout}>Sair</MenuItem>
    </Menu>
  );

  const mobileMenuId = 'primary-search-account-menu-mobile';
  const renderMobileMenu = (
    <Menu
      anchorEl={mobileMoreAnchorEl}
      anchorOrigin={{
        vertical: 'top',
        horizontal: 'right',
      }}
      id={mobileMenuId}
      keepMounted
      transformOrigin={{
        vertical: 'top',
        horizontal: 'right',
      }}
      open={isMobileMenuOpen}
      onClose={handleMobileMenuClose}
    >
      <MenuItem onClick={handleProfileMenuOpen}>
        <IconButton
          aria-label="conta do usuário"
          aria-controls="primary-search-account-menu"
          aria-haspopup="true"
          color="inherit"
        >
          <AccountCircle />
        </IconButton>
        <p>Profile</p>
      </MenuItem>
    </Menu>
  );

  return (
    <FlexGrow>
      <AppBar position="static">
        <Toolbar>
          <MenuButton edge="start" color="inherit" aria-label="open drawer">
            <MenuIcon />
          </MenuButton>
          <Link href="/">
            <LogoWrapper>
              <LogoImg src="/pdv-logo-white.png" />
            </LogoWrapper>
          </Link>
          <FlexGrow />
          {/* <Search>
            <SearchIconWrapper>
              <SearchIcon />
            </SearchIconWrapper>
            <CustomInput
              placeholder="Buscar..."
              inputProps={{ 'aria-label': 'buscar' }}
            />
          </Search> */}
          <DesktopSection>
            <IconButton
              edge="end"
              aria-label="conta do usuário"
              aria-controls={menuId}
              aria-haspopup="true"
              onClick={handleProfileMenuOpen}
              color="inherit"
            >
              <AccountCircle />
            </IconButton>
          </DesktopSection>
          <MobileSection>
            <IconButton
              aria-label="mostrar mais"
              aria-controls={mobileMenuId}
              aria-haspopup="true"
              onClick={handleMobileMenuOpen}
              color="inherit"
            >
              <MoreIcon />
            </IconButton>
          </MobileSection>
        </Toolbar>
      </AppBar>
      {renderMobileMenu}
      {renderMenu}
    </FlexGrow>
  );
};

export default HeaderBar;
