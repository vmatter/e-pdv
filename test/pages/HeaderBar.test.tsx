import { render } from '../testUtils';
import HeaderBar from '../../components/HeaderBar';

describe('HeaderBar', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<HeaderBar />, {});
    expect(asFragment()).toMatchSnapshot();
  });

  test('User account', async () => {
    const { getByText } = render(<HeaderBar />);

    expect(getByText('Usu√°rios')).toBeInTheDocument();

    expect(getByText('Produtos')).toBeInTheDocument();

    expect(getByText('Sair')).toBeInTheDocument();
  });
});
