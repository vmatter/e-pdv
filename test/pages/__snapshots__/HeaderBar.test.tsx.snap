// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-19op8cy-FlexGrow e9powp66"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-eyxyjd-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <a
          class="css-3ahs8u-LogoWrapper e9powp64"
        >
          <img
            class="css-q33v58-LogoImg e9powp63"
            src="/pdv-logo-white.png"
          />
        </a>
        <div
          class="css-19op8cy-FlexGrow e9powp66"
        />
        <div
          class="css-odgcyt-DesktopSection e1sq2t0a1"
        >
          <button
            aria-controls="primary-search-account-menu"
            aria-haspopup="true"
            aria-label="conta do usuÃ¡rio"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-e74czq-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label css-4jkopv-MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AccountCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="css-n5dn9s-MobileSection e1sq2t0a0"
        >
          <button
            aria-controls="primary-search-account-menu-mobile"
            aria-haspopup="true"
            aria-label="mostrar mais"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-1drqkcs-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label css-4jkopv-MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="MoreVertIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
